UM/Math/Polygon.py

```diff
@@ -175,8 +175,10 @@ def intersectsPolygon(self, other: "Polygon") -> Optional[Tuple[float, float]]:
 
         polygon_me = ShapelyUtil.polygon2ShapelyPolygon(self)
         polygon_other = ShapelyUtil.polygon2ShapelyPolygon(other)
-
-        if not (polygon_me.is_valid and polygon_other.is_valid):
+        if not polygon_other.is_empty and not polygon_me.is_empty:
+            if not polygon_me.is_valid and polygon_other.is_valid:  # If not valid
+                return None
+        else:
             return None
```

Comment:
For readability, it's perhaps easier to just write down the two cases in which we want to return `None`, like so:

    if not (polygon_me.is_valid and polygon_other.is_valid):  # If not valid
        return None
    if polygon_other.is_empty or polygon_me.is_empty:
        return None

It's a lot easier to follow the semantics of the code, that way, don't you think?

I'm sorry for all the confusion and corrections.

Reply:
@Ghostkeeper 

My intention is to execute "polygon_me.is_valid" AFTER is_empty check.(So it won't throw null - exception)

I pretty much sure that both(your suggestion/mine) works the same. Will change it after comment.